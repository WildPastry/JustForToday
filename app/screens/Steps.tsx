import { ColorSchemeName, StyleSheet } from 'react-native';
import { ForwardedScrollView, View } from '../components/styles/Themed';
import React, { useRef } from 'react';
import { AppState } from '../redux/store';
import Colours from '../constants/Colours';
import Control from '../constants/Control';
import { FontDisplay } from '../components/styles/StyledText';
import { IDeviceSize } from '../types/generic.types';
import { IStep } from '../types/data.types';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import Step from '../components/Step';
import getDeviceSize from '../constants/Layout';
import { useAppSelector } from '../redux/hooks';
import useColorScheme from '../hooks/useColorScheme';
import { useFocusEffect } from '@react-navigation/native';

const Steps: React.FC = (): JSX.Element => {
  // Screen settings
  const scrollViewRef: React.MutableRefObject<any> = useRef<any>(null);
  const colorScheme: NonNullable<ColorSchemeName> = useColorScheme();
  const deviceSize: IDeviceSize[keyof IDeviceSize] = getDeviceSize();

  // Data from store
  const steps: IStep[] = useAppSelector((state: AppState): IStep[] => {
    return state.data.steps;
  });

  useFocusEffect(
    React.useCallback(() => {
      // Scroll to top on focus
      scrollViewRef.current?.scrollTo({ y: 0, animated: false });
      return () => null;
    }, [scrollViewRef])
  );

  return (
    <ForwardedScrollView
      contentContainerStyle={Control[deviceSize].container}
      ref={scrollViewRef}>
      <View style={styles.logoContainer}>
        {/* Logo */}
        <MaterialCommunityIcons
          style={[styles.icon, { marginRight: Control[deviceSize].iconMargin }]}
          name='stairs'
          size={Control[deviceSize].icon}
          color={Colours[colorScheme].text}
        />
        {/* Title */}
        <FontDisplay style={Control[deviceSize].title}>Steps</FontDisplay>
      </View>
      {/* Divider */}
      <View
        style={styles.divider}
        lightColor={Colours.light.seperator}
        darkColor={Colours.dark.seperator}
      />
      {/* Steps */}
      {steps.map((step, index) => (
        <Step key={index} id={step.id} step={step.step} />
      ))}
    </ForwardedScrollView>
  );
};

const styles = StyleSheet.create({
  logoContainer: {
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'center',
    marginTop: 10
  },
  icon: {
    textAlign: 'center'
  },
  divider: {
    alignSelf: 'center',
    height: 1,
    marginBottom: 20,
    marginTop: 30,
    width: '70%'
  }
});

export default Steps;
