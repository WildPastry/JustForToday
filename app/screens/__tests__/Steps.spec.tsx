import { Provider } from 'react-redux';
import Steps from '../Steps';
import { makeStore } from '../../redux/store';
import renderer from 'react-test-renderer';

describe('<Steps />', () => {
  it('renders correctly', () => {
    const store = makeStore();
    const tree = renderer
      .create(
        <Provider store={store}>
          <Steps />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
