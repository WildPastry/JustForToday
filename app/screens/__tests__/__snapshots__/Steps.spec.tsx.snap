// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Steps /> renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "padding": 20,
    }
  }
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
>
  <View>
    <View
      style={
        [
          {
            "backgroundColor": "#fff",
          },
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 10,
          },
        ]
      }
    >
      <Text />
      <Text
        style={
          [
            {
              "color": "#000",
            },
            [
              {
                "fontSize": 22,
                "marginLeft": 10,
              },
              {
                "fontFamily": "font-display",
              },
            ],
          ]
        }
      >
        Steps
      </Text>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "#bdc3c7",
          },
          {
            "alignSelf": "center",
            "height": 1,
            "marginBottom": 30,
            "marginTop": 20,
            "width": "70%",
          },
        ]
      }
    />
    <Text
      style={
        [
          {
            "color": "#000",
          },
          {
            "flexDirection": "row",
            "flexWrap": "wrap",
            "marginBottom": 20,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#000",
            },
            {
              "fontSize": 15,
              "fontWeight": "200",
              "lineHeight": 21,
            },
          ]
        }
      >
        1
          
      </Text>
      <Text
        style={
          [
            {
              "color": "#000",
            },
            {
              "fontSize": 15,
              "lineHeight": 21,
              "textAlign": "left",
            },
          ]
        }
      >
        Step 1
      </Text>
    </Text>
    <Text
      style={
        [
          {
            "color": "#000",
          },
          {
            "flexDirection": "row",
            "flexWrap": "wrap",
            "marginBottom": 20,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#000",
            },
            {
              "fontSize": 15,
              "fontWeight": "200",
              "lineHeight": 21,
            },
          ]
        }
      >
        2
          
      </Text>
      <Text
        style={
          [
            {
              "color": "#000",
            },
            {
              "fontSize": 15,
              "lineHeight": 21,
              "textAlign": "left",
            },
          ]
        }
      >
        Step 2
      </Text>
    </Text>
  </View>
</RCTScrollView>
`;
